broker/lib/jobs/ServiceInstanceUpdateJob.js

```diff
@@ -97,7 +97,13 @@ class ServiceInstanceUpdateJob extends BaseJob {
           })
           .catch(err => {
             operationResponse.update_init = CONST.OPERATION.FAILED;
-            logger.error('Error occurred while updating service insance job :', err);
+            logger.error('Error occurred while updating service instance job :', err);
+            if (err instanceof errors.DeploymentAttemptRejected) {
+              //If deployment was staggered due to exhaustion of workers, reschedule update job
+              //Retry attempts do not count when deployment is staggered
+              trackAttempts = false;
+              err.statusMessage = 'Deployment attempt rejected due to BOSH overload. Update cannot be initiated';
+            }
```

Comment:
The update retry is unbounded here but it would be a good idea to have a timeout based on the firstAttempt started at. And if the update run is after the next scheduled update run (ie. after 7 days), then dont have to retry because then you would have two competing runs on the same deployment and it might simply vie for the queue.

For now you might not need to really fix this as it might simply make the code a bit complex here. Just document a TODO and after extensive tests in canary if we need to fix this, lets patch it.